hibernate-core/src/main/java/org/hibernate/engine/internal/StatefulPersistenceContext.java

```diff
@@ -684,6 +684,7 @@ public Object narrowProxy(Object proxy, EntityPersister persister, EntityKey key
 
 			// Otherwise, create the narrowed proxy
 			final HibernateProxy narrowedProxy = (HibernateProxy) persister.createProxy( key.getIdentifier(), session );
```

Comment:
@joshlandin, I'm not sure of the right thing to do here. Let's see what @sebersole thinks.